### DISCUSSION TITLE
No documentation about *_STATUS_SEGMENTS variable?

### DISCUSSION THREAD

**Discussion by davidshen84** [erikw/tmux-powerline#441] [Q&A] [answered]
Hi,

I am new to this tool. I am tryingt to customize my powerline status bar to display the k8s context.

I updated the code in the `kubernetes_context.sh` section in my powerline `config.sh` file, but did not see any change in my powerline bar.

After much debugging and investigating, I found I need to update these variables to enable or disable the segments:

- TMUX_POWERLINE_LEFT_STATUS_SEGMENTS
- TMUX_POWERLINE_RIGHT_STATUS_SEGMENTS

But there's document mention them. Am I supposed to tinker with them?

What would be the right way to enable this segment?

Also, I want to remove the battery and I too find I have to update these variables to achieve that.

Thanks.

**Comment by davidshen84** [comment]
ok...I think they are not the right place to customize the segments. After I set the values, the background color of the status bar is removed.

**Comment by xx4h** [comment, marked as answer]
Hi there,

i agree, that we're lacking some documentation regarding how to configure segments! I'll put that on the list for when i have some more time in the upcoming weeks.

But in general, you have two ways of configuring the segments you want to use:

# Configure segments in your config
Copying them from the theme you currently use is probably a good starting point, so if you're using the default theme, i'd copy the following lines to the top of my `config.sh`

https://github.com/erikw/tmux-powerline/blob/717688b1dd9ca3bf062b44a59fb2206e07e28418/themes/default.sh#L104-L118
and
https://github.com/erikw/tmux-powerline/blob/717688b1dd9ca3bf062b44a59fb2206e07e28418/themes/default.sh#L123-L141

**Don't forget to prepend `export` to both `TMUX_POWERLINE_LEFT_STATUS_SEGMENTS` and `TMUX_POWERLINE_RIGHT_STATUS_SEGMENTS`**

# Create your own theme
Copy the default theme and modify the theme itself:
```
mkdir -p ~/.config/tmux-powerline/themes
cp -a ~/.tmux/plugins/tmux-powerline/themes/default.sh ~/.config/tmux-powerline/themes/mytheme.sh

# modify
vim ~/.config/tmux-powerline/themes/mytheme.sh

# set your new theme in your config, set TMUX_POWERLINE_THEME="mytheme"
vim ~/.config/tmux-powerline/config.sh
```

**Reply by davidshen84** [reply]
Thanks a lot.

I think copying that two variables to `config.sh` won't work in most cases. Because most themes, even the default one, uses variables, like `TMUX_POWERLINE_SEPARATOR_LEFT_THIN`, and they are not available in the config.sh file. Unless you copy them all alone.

Using a default theme as a base and modify it seems easier.
